src/git.rs

```diff
@@ -65,7 +64,37 @@ impl GitConfig {
     }
 }
 
-pub(crate) fn create(project_dir: &Path, args: GitConfig) -> Result<String> {
+fn create_with_ssh(project_dir: &Path, args: GitConfig) -> Result<String> {
+    let mut callbacks = RemoteCallbacks::new();
+
+    callbacks.credentials(|_url, username_from_url, _allowed_types| {
+        Cred::ssh_key(
+            username_from_url.unwrap_or("git"),
+            None,
+            Path::new(&format!("{}/.ssh/id_rsa", env::var("HOME").unwrap())),
+            None,
+        )
+    });
+
+    let branch = match args.branch {
```

Comment:
thought: this code, specifically the fallback for the default branch name to `"master"` differs a bit bit from the code that is used in case of a http url. (see Line 119 - 123). It would be very good to reuse this code as much as possible to have a sane, consistent behaviour for both sort of urls.

Reply:
suggestion: I think it would be simpler to have just one function for the `create` as it was before, and do a little pre ceremony for the credentials in case the url is a ssh url. That way you would basically depend literally on the very same code and just to the things that are needed for the other url type upfront and then the rest should not differ at all.